### DISCUSSION TITLE
Setting File Name?

### DISCUSSION THREAD

**Discussion by maribox** [typst/typst#3957] [Q&A] [answered]
Hello,
I am currently using the web app to write documents that are always named after variables I have saved inside the document. Is there any way I can set the name of the downloaded document, or when using the compiler myself, set the name based on variables inside the document without parsing it myself? Or would I need to write a wrapper into which I pass the variables then?

**Comment by leggewie** [comment, marked as answer]
you need to write a wrapper.  please share it if you do. I believe this is entirely impossible with the publicly hosted typst app.

https://forum.typst.app/t/can-a-typ-file-set-the-output-pdf-file-name/1009
